### PR TITLE
docs: Add more Gateway deploy examples

### LINKED ISSUE #3041: Docker compose Gateway deployment
Customer contributed a sample working `docker-compose.yml` to deploy a gateway.

- [ ] Add relevant docs
- [ ] Add upgrade docs
- [ ] Add relevant deployment instructions in portal

```yaml
version: '3.7'

services:
  firezone-gateway:
    image: ghcr.io/firezone/gateway:1
    environment:
      - FIREZONE_ID=<id>
      - FIREZONE_TOKEN=<token>
      - RUST_LOG=str0m=warn,info
      - FIREZONE_ENABLE_MASQUERADE=1
      - FIREZONE_NAME=<name-of-gateway>
    cap_add:
      - NET_ADMIN
    sysctls:
      - net.ipv4.conf.all.src_valid_mark=1
      - net.ipv6.conf.all.disable_ipv6=0
      - net.ipv6.conf.all.forwarding=1
      - net.ipv6.conf.default.forwarding=1
    healthcheck:
      test: ["CMD", "ip", "link", "|", "grep", "tun-firezone"]
      interval: 5s
      timeout: 10s
      retries: 3
      start_period: 1m
    devices:
      - /dev/net/tun:/dev/net/tun
```

### LINKED ISSUE #5072: Show instructions in docs for deploying Gateways for different infra
Add a `/kb/automate` section with more examples.

### PR DESCRIPTION
**Pull request by jamilbk**
- ~~Adds `fontawesome` icons to the portal and integrates them into the existing `icon` component. Use just like the heroicons except prefixed `fa-`.~~ Edit: removed these because I don't think it's worth the increased bundle size. Just going to inline the icons we need instead.
- Fixes #3041
- Fixes #5072

### CODE REVIEW

File: elixir/apps/web/lib/web/components/core_components.ex [hunk 1]
```diff
@@ -152,13 +153,18 @@ defmodule Web.CoreComponents do
...
             <li class="mr-2" role="presentation">
               <button
-                class={[
-                  (Map.get(tab, :selected) && "rounded-t text-accent-600 border-accent-600") ||
-                    "text-neutral-500 hover:border-accent-200 hover:text-accent-600",
-                  "inline-block p-4 border-b-2"
-                ]}
+                class={
+                  [
+                    # ! is needed to override Flowbite's default styles
```
Comment (jamilbk) [inline elixir/apps/web/lib/web/components/core_components.ex:161]: Fixes colors in the tabs component.

File: elixir/apps/web/lib/web/live/sites/new_token.ex [hunk 2]
```diff
@@ -78,11 +74,12 @@ defmodule Web.Sites.NewToken do
           <.tabs :if={@env} id="deployment-instructions">
             <:tab
               id="systemd-instructions"
+              icon="hero-command-line"
               label="systemd"
               phx_click="tab_selected"
               selected={@selected_tab == "systemd-instructions"}
             >
-              <p class="p-4">
+              <p class="p-6">
```
Comment (jamilbk) [inline elixir/apps/web/lib/web/live/sites/new_token.ex:82]: This aligns the content of the tabs component with the content inside the code blocks

**Comment by jamilbk** [comment]
bump @bmanifold

File: elixir/apps/web/lib/web/components/core_components.ex [hunk 3]
```diff
@@ -130,6 +130,7 @@ defmodule Web.CoreComponents do
   slot :tab, required: true, doc: "Tab content" do
     attr :id, :string, required: true, doc: "ID of the tab"
     attr :label, :any, required: true, doc: "Display label for the tab"
+    attr :icon, :string, doc: "Icon for the tab"
```
Comment (bmanifold) [inline elixir/apps/web/lib/web/components/core_components.ex:133]: 👍

File: website/src/app/kb/automate/docker-compose/readme.mdx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+   deployments.
+1. [Compose plugin for Docker Engine](https://docs.docker.com/compose/install/#scenario-two-install-the-compose-plugin)
+   version 2.0.0 or later. We recommend using the latest version available.
+
+## Deployment
+
+Use the following `docker-compose.yml`, customizing as needed:
+
+```yaml
+version: "3.7"
```
Comment (bmanifold) [inline website/src/app/kb/automate/docker-compose/readme.mdx]: Not a huge deal, but `version` has been deprecated
https://docs.docker.com/compose/compose-file/04-version-and-name/#version-top-level-element-obsolete

File: website/src/components/KbSidebar/index.tsx [hunk 5]
```diff
@@ -155,5 +158,6 @@ export default function KbSidebar() {
         </SidebarItemGroup>
       </SidebarItems>
     </Sidebar>
+>>>>>>> main
```
Comment (bmanifold) [inline website/src/components/KbSidebar/index.tsx]: I'm guessing this was left in by mistake.

**Review by bmanifold — COMMENTED** [review]
Overall, looks good.  I think there is a git merge conflict comment that was left in and causing the site to not be able to deploy on Vercel.

File: website/src/app/kb/automate/docker-compose/readme.mdx (hunk 4 above)
Comment (jamilbk) [reply website/src/app/kb/automate/docker-compose/readme.mdx]: 👍

File: website/src/components/KbSidebar/index.tsx (hunk 5 above)
Comment (jamilbk) [reply website/src/components/KbSidebar/index.tsx]: Whoops, removed.

**Review by bmanifold — APPROVED** [review]

**Pull request merged**